controllers/apps/operations/horizontal_scaling.go

```diff
@@ -447,3 +460,152 @@ func (hs horizontalScalingOpsHandler) getCompExpectedOfflineInstances(
 	}
 	return compOfflineInstances
 }
+
+// HandleAbortForHScale handling the aborted horizontal scaling.
+func HandleAbortForHScale(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) error {
+	var (
+		needUpdateCluster bool
+		err               error
+		ops               = opsRes.OpsRequest
+		opsDeepCopy       = ops.DeepCopy()
+	)
+	if ops.Annotations[constant.RetryAbortedHScaleAnnoKey] == "true" {
+		if needUpdateCluster, err = handleClusterCompSpecForAbortedHScale(opsRes, appsv1alpha1.AbortedProgressStatus,
+			appsv1alpha1.PendingProgressStatus, true); needUpdateCluster {
```

Comment:
It seems if ops.Annotations[constant.RetryAbortedHScaleAnnoKey] is true, handleClusterCompSpecForAbortedHScale will "reverse" the operation.

reads counter-intuitive.